// evergreen-ui/commonjs/menu/src/MenuOptionsGroup.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _layers = require("../../layers");

var _typography = require("../../typography");

var _theme = require("../../theme");

var _MenuOption = _interopRequireDefault(require("./MenuOption"));

var MenuOptionsGroup =
/*#__PURE__*/
function (_React$PureComponent) {
  (0, _inherits2.default)(MenuOptionsGroup, _React$PureComponent);

  function MenuOptionsGroup() {
    (0, _classCallCheck2.default)(this, MenuOptionsGroup);
    return (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(MenuOptionsGroup).apply(this, arguments));
  }

  (0, _createClass2.default)(MenuOptionsGroup, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          title = _this$props.title,
          options = _this$props.options,
          selected = _this$props.selected,
          onChange = _this$props.onChange;
      return _react.default.createElement(_layers.Pane, {
        paddingY: 8
      }, title && _react.default.createElement(_typography.Heading, {
        size: 100,
        marginLeft: 44,
        marginRight: 16,
        marginY: 8
      }, title), _react.default.createElement(_layers.Pane, null, options.map(function (option) {
        return _react.default.createElement(_MenuOption.default, {
          key: option.value,
          isSelected: option.value === selected,
          onSelect: function onSelect() {
            return onChange(option.value);
          }
        }, option.label);
      })));
    }
  }]);
  return MenuOptionsGroup;
}(_react.default.PureComponent);

MenuOptionsGroup.displayName = "MenuOptionsGroup";
(0, _defineProperty2.default)(MenuOptionsGroup, "propTypes", {
  /**
   * Title of the menu group.
   */
  title: _propTypes.default.node,

  /**
   * The current value of the option group.
   */
  selected: _propTypes.default.any,

  /**
   * Function called when selection changes.
   */
  onChange: _propTypes.default.func,

  /**
   * List of options rendered in the group.
   */
  options: _propTypes.default.array
});

var _default = (0, _theme.withTheme)(MenuOptionsGroup);

exports.default = _default;
